cpp/src/parquet/chunker_internal_test.cc

```diff
@@ -0,0 +1,908 @@
+                      const std::vector<uint64_t>& original,
+                      const std::vector<uint64_t>& modified, uint8_t n_modifications,
+                      uint64_t edit_length) {
+  auto diffs = FindDifferences(original, modified);
+  if (diffs.size() != n_modifications) {
+    PrintDifferences(original, modified, diffs);
+  }
+  ASSERT_EQ(diffs.size(), n_modifications);
+
+  for (const auto& diff : diffs) {
+    if (!::arrow::is_list_like(dtype->id())) {
+      uint64_t left_sum = 0, right_sum = 0;
+      for (const auto& val : diff.first) left_sum += val;
+      for (const auto& val : diff.second) right_sum += val;
+      ASSERT_EQ(left_sum + edit_length, right_sum);
+    }
+    ASSERT_LE(diff.first.size(), 2);
+    ASSERT_LE(diff.second.size(), 2);
+  }
+}
+
+void AssertAppendCase(const std::vector<uint64_t>& original,
+                      const std::vector<uint64_t>& modified) {
+  ASSERT_GE(modified.size(), original.size());
+  for (size_t i = 0; i < original.size() - 1; i++) {
+    ASSERT_EQ(original[i], modified[i]);
+  }
+  ASSERT_GT(modified[original.size() - 1], original.back());
+}
+
+uint64_t ElementCount(uint64_t size, int32_t byte_width, bool nullable) {
+  if (nullable) {
+    // in case of nullable types the def_levels are also fed through the chunker
+    // to identify changes in the null bitmap, this will increase the byte width
+    // and decrease the number of elements per chunk
+    byte_width += 2;
+  }
+  return size / byte_width;
+}
+
+void AssertChunkSizes(const std::shared_ptr<::arrow::DataType>& dtype,
+                      PageSizes base_result, PageSizes modified_result, bool nullable,
+                      bool enable_dictionary, uint64_t min_chunk_size,
+                      uint64_t max_chunk_size) {
+  max_chunk_size = static_cast<uint64_t>(max_chunk_size * 1.2);
+  if (::arrow::is_fixed_width(dtype->id())) {
+    auto min_length = ElementCount(min_chunk_size, dtype->byte_width(), nullable);
+    auto max_length = ElementCount(max_chunk_size, dtype->byte_width(), nullable);
+    AssertAllBetween(base_result.lengths, min_length, max_length,
+                     /*expect_dictionary_fallback=*/enable_dictionary);
+    AssertAllBetween(modified_result.lengths, min_length, max_length,
+                     /*expect_dictionary_fallback=*/enable_dictionary);
+  } else if (::arrow::is_base_binary_like(dtype->id()) && !nullable &&
+             !enable_dictionary) {
+    AssertAllBetween(base_result.sizes, min_chunk_size, max_chunk_size);
+    AssertAllBetween(modified_result.sizes, min_chunk_size, max_chunk_size);
+  }
+}
+
+constexpr uint64_t kMinChunkSize = 8 * 1024;
+constexpr uint64_t kMaxChunkSize = 32 * 1024;
+constexpr uint64_t kPartSize = 128 * 1024;
+constexpr uint64_t kEditSize = 128;
+
+class TestColumnCDC : public ::testing::TestWithParam<
+                          std::tuple<std::shared_ptr<::arrow::DataType>, bool, size_t>> {
```

Comment:
Can the `std::tuple` be replace with a well-named struct?

Reply:
Updated to use a struct.